proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,74 @@
+# NuGet configuration CLI for dotnet.exe
+
+- Author: [Heng Liu](https://github.com/heng-liu)
+- GitHub Issue [8420](https://github.com/NuGet/Home/issues/8420)
+
+## Problem Background
+
+Currently, there is no NuGet configuration CLI for dotnet.exe. It's inconvenient for NuGet users to know about the NuGet configuration file locations and figure out where is the merged configuration coming from. 
+
+## Who are the customers
+
+This feature is for dotnet.exe users.
+
+## Goals
+Design and implement `dotnet nuget config` command.
+
+## Non-Goals
+Design and implement `dotnet nuget config` command with commands other than `list`, E.g. add/update/delete
+
+## Solution
+The following command will be implemented in the `dotnet.exe` CLI.
+
+### `dotnet nuget config`
+
+#### Commands
+If no command is specified, the command will default to list.
+
+- List
+
+Lists all the NuGet configuration file locations. This command will include all the NuGet configuration file that will be applied, when invoking NuGet command from the current working directory path. The listed NuGet configuration files are in priority order. So the order of loading those configurations is reversed, that is, loading order is from the bottom to the top. So the configuration on the top will apply.
+You may refer to [How settings are applied](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#how-settings-are-applied) for more details. 
+
+#### Options
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+- --current-directory
+
+Run this command as if current directory is set to the specified directory.
+
+#### Examples
+
+- List all the NuGet configuration file that will be applied, when invoking NuGet command in the current directory.
+
+```
+dotnet nuget config list
+
+c:\repos\Solution\Project\NuGet.Config
```

Comment:
We discourage customers trying to nuget.config files in project directories, because VS will never use it, and I'm not sure that command line solution restore will either. Therefore, we shouldn't use discourages practices in our examples.

Reply:
Thanks for letting me know! Changed the examples accordingly.